python/lsst/ts/atwhitelight/atwhitelight_csc.py

```diff
@@ -0,0 +1,446 @@
+            await self.disconnect_lamp()
+            await self.disconnect_chiller()
+
+    async def connect_chiller(self):
+        await self.disconnect_chiller()
+        if self.chiller_model is None:
+            self.chiller_model = ChillerModel(
+                config=self.config.chiller,
+                topics=self,
+                log=self.log,
+                status_callback=self.status_callback,
+                simulate=self.simulation_mode != 0,
+            )
+        await self.chiller_model.connect()
+
+    async def connect_lamp(self):
+        await self.disconnect_lamp()
+        if self.lamp_model is None:
+            labjack_config = self.config.lamp
+            self.lamp_model = LampModel(
+                config=labjack_config,
+                topics=self,
+                log=self.log,
+                status_callback=self.status_callback,
+                simulate=self.simulation_mode != 0,
+            )
+        await self.lamp_model.connect()
+
+    async def disconnect_chiller(self):
+        # Don't use chiller_connected because that can be false
+        # even if a basic connection exists (before status seen)
+        if self.chiller_model is None:
+            return
+        await self.chiller_model.disconnect()
+
+    async def disconnect_lamp(self):
+        # Don't use lamp_connected because that can be false
+        # even if a basic connection exists (before status seen)
+        if self.lamp_model is None:
+            return
+        await self.lamp_model.disconnect()
+
+    async def begin_standby(self, data):
+        """Make sure the hardware is not reporting errors"""
+        # don't let the user leave fault state if the KiloArc
+        # or chiller is reporting an error
+        if self.summary_state == salobj.State.FAULT:
+            if self.lamp_connected:
+                lamp_state = self.lamp_model.get_state()
+                if lamp_state.controllerError != LampControllerError.NONE:
+                    raise salobj.ExpectedError(
+                        f"Lamp controller error code={lamp_state.controllerError!r}"
+                    )
+            if self.chiller_connected:
+                chiller_watchdog = self.chiller_model.get_watchdog()
+                if chiller_watchdog.alarmsPresent:
+                    raise RuntimeError(
+                        "Can't go to standby: chiller is reporting one or more alarms"
+                    )
+        if self.lamp_connected:
+            remaining_cooldown = self.lamp_model.get_remaining_cooldown()
+            if remaining_cooldown > 0:
+                raise salobj.ExpectedError(
+                    "The lamp is cooling. You can't go to standby "
+                    "until the chiller has run for another "
+                    f"{remaining_cooldown:0.2f} seconds."
+                )
+
+    async def start(self):
+        await self.evt_lampConnected.set_write(connected=False)
```

Comment:
Perhaps move that to begin_start? Will save you need to await super().start(), and might provide better fault handling.

Reply:
They are very different (as you probably know). The start method is just called at startup. After that I would prefer to rely on the CSC code to report the correct connected state.

(By the way: I have to call super().begin_start() if I override that; ConfigurableCsc uses it.)